src/test/util_tests.cpp

```diff
@@ -76,6 +76,9 @@ BOOST_AUTO_TEST_CASE(util_check)
     const int two = *Assert(p_two);
     Assert(two == 2);
     Assert(true);
+    // Check that Assume can be used as unary expression
+    const bool result{Assume(two == 2)};
+    Assert(result);
```

Comment:
fun fact: I tried to minimize the test by writing `Assert(Assume(two == 2))`, but that only works in [C++20](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2017/p0315r4.pdf).

Reply:
The added lines are intended to test `Assume`, therefore we could s/`Assert(result);`/`BOOST_CHECK(true);`/ to suppress "Test case util_tests/util_check did not check any assertions" in
```
$ ./src/test/test_bitcoin -t util_tests/util_check -l test_suite
Running 1 test case...
Entering test module "Bitcoin Core Test Suite"
test/util_tests.cpp(48): Entering test suite "util_tests"
test/util_tests.cpp(75): Entering test case "util_check"
Test case util_tests/util_check did not check any assertions
test/util_tests.cpp(75): Leaving test case "util_check"; testing time: 9563us
test/util_tests.cpp(48): Leaving test suite "util_tests"; testing time: 9634us
Leaving test module "Bitcoin Core Test Suite"; testing time: 9695us

*** No errors detected
```

If `Assert(result);` is really required (but I cannot see reasons), maybe add `BOOST_CHECK(true);` as it is done here: https://github.com/bitcoin/bitcoin/blob/7efc628539573af4b4a76d93b853cc46e9e52eae/src/test/util_tests.cpp#L102 ?

Reply:
`BOOST_CHECK(result);` should also work

Reply:
Right. I meant it :) A copy-paste error...